libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFSDKOAuth2.h

```diff
@@ -89,8 +89,8 @@ NS_ASSUME_NONNULL_BEGIN
 @property (nonatomic, readonly) NSString *accessToken;
 @property (nonatomic, readonly) NSString *refreshToken;
 @property (nonatomic, readonly) NSDate *issuedAt;
-@property (nonatomic, readonly) NSURL *instanceUrl;
-@property (nonatomic, readonly) NSURL *identityUrl;
+@property (nonatomic, readonly, nullable) NSURL *instanceUrl;
```

Comment:
If I'm remembering right, these were intended to be nonnull but in that case we could fix the implementations so they don't have a nil return, @trooper2013 would know better :)

Reply:
Thanks @bbirman! @trooper2013 could you provide tips here? Should I keep this `nonnull` and fix the implementation?

Reply:
Yes, this should be `nonnull`.

Reply:
@bhariharan If `instanceUrl` is designed to be `nonnull`, then we should fix the implementation:

```objc
- (NSURL *)instanceUrl {
    if (self.values[kSFOAuthInstanceUrl]) {
        return [NSURL URLWithString:self.values[kSFOAuthInstanceUrl]];
    }
    return nil;
}
```

Any tip on what should it return if `kSFOAuthInstanceUrl` is not found in `values` dictionary?

Reply:
Remove the check, and directly return `[NSURL URLWithString:self.values[kSFOAuthInstanceUrl]]`. Instance URL should always be present.